### ISSUE TITLE
Segmentation Fault   - Raspberry Pi 3 and Jetson TX1

### ISSUE THREAD

**Issue by smajida** [davisking/dlib#760]
Hi,

I am going to install Dlib-19.4 from source on a Raspberry Pi 3. Compiling Dlib itself and its Python API were not a problem. However, when I want to run unit test suite after running `./dtest --runall`, I get the following message after a couple of minutes

`Running test_matrix3 | Segmentation Fault`.

I have already increased swap file to 2048MB and decreased Memory split to 16. Also booting is on the console and not desktop.

**Comment by davisking** [comment]
Maybe your blas library installed on your pi is broken.  Try compiling dlib without blas. You can disable blas via the cmake option.

**Comment by smajida** [comment]
@davisking 
Isn't blas library needed for face detection and recognition?
I also tried the same with Jetson TX1 and got the following issue:
`Running test_cublas | Segmentation fault`

**Comment by davisking** [comment]
No, BLAS just makes things faster.  But lots of systems have incorrectly installed or otherwise broken BLAS libraries.

**Comment by smajida** [comment]
I doubled check again to make sure BLAS library was installed correctly.
`(facerec-cv2.13.4-py2.7) nvidia@tegra-ubuntu:~/projects/face-recognition/softwares/dlib-master/dlib/test/build/examples/examples_build$ sudo apt-get install lib
blas-dev liblapack-dev --upgrade
[sudo] password for nvidia: 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
libblas-dev is already the newest version (3.6.0-2ubuntu2).
liblapack-dev is already the newest version (3.6.0-2ubuntu2).
0 upgraded, 0 newly installed, 0 to remove and 12 not upgraded.
`
as mentioned before following compilation was done with no error. Except AVX_INSTRUCTION is not suppored in Jetson TX1 according to message I saw after executing the following commands:

`mkdir build; cd build; cmake .. -DARM_NEON=1 -DUSE_AVX_INSTRUCTIONS=1; cmake --build .`

`python setup.py install --yes USE_AVX_INSTRUCTIONS --compiler-flags "-O3 -fprofile-use"`

However, still it is not possible to see output of face detection. Every time when I try to see the demo of face detection by Deep Learning methods, one window is opened with "question mark" in toolbar panel, but no window is opened.Then after 5 seconds it shows the program is "Killed!"

`(facerec-cv2.13.4-py2.7) nvidia@tegra-ubuntu:~/projects/face-recognition/softwares/dlib-master/dlib/test/build/examples/examples_build$ ./dnn_mmod_face_detection_ex mmod_human_face_detector.dat ../../../../../examples/faces/2008_002079.jpg 
Killed
`

**Comment by davisking** [comment]
You are probably running out of RAM.  Make the input image smaller.

**Comment by smajida** [comment]
I investigated this issue. I gets close to RAM memory limit, but it does not hit the maximum. I am surprised as before executing the command only 1.1 GB RAM out of 4 GB RAM memory is occupied. 

I tried `2008_001322.jpg` image in faces folder which is ony 500x375px, but it still did not work. even smaller than this size did not resolve the issue. I could not find anyone has ever tried dnn example of face detection on Jetson TX1 to get some hints.

![screenshot from 2017-08-23 19-48-56](https://user-images.githubusercontent.com/12903362/29630381-5482533c-883c-11e7-801d-c64ea0592ace.png)

**Comment by e-fominov** [comment]
@smajida , you really don't have enough RAM to run this example. As I see on your images, you have only 2.8 GB available. 23 seconds ago there were a peak of memory usage (90%) - I think the example was running this time and it was not able to allocate required amount of memory

[The reason are this lines of code inside example](https://github.com/davisking/dlib/blob/master/examples/dnn_mmod_face_detection_ex.cpp#L91-L92):

```C++
        while(img.size() < 1800*1800)
            pyramid_up(img);
```

I recommend you to comment this lines and try to recompile and run example

**Comment by smajida** [comment]
thanks @e-fominov 

I have commented out that part. I tried with one of the images in faces folder, however it resulted in no detected faces.
![screenshot from 2017-08-29 02-18-42](https://user-images.githubusercontent.com/12903362/29799023-34edfe4e-8c61-11e7-84a5-e205d6be1b41.png)

@davisking has mentioned faces should be at least 80x80. It reaches me to this conclusion that on Jetson TX1, faces in images like this one can not be applied for face detection using CNN Dlib model.
